yeat/cli/yeat_auto.py

```diff
@@ -0,0 +1,60 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2024, DHS. This file is part of YEAT: http://github.com/bioforensics/yeat
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+
+from argparse import ArgumentParser
+from yeat.config.auto_pop import AutoPop
+
+
+def main(args=None):
+    if args is None:
+        args = get_parser().parse_args()  # pragma: no cover
+    autopop = AutoPop(args.samples, args.seq_path, args.files)
+    autopop.write_config_file(args.outfile)
+
+
+def get_parser(exit_on_error=True):
+    parser = ArgumentParser(exit_on_error=exit_on_error)
+    options(parser)
+    positional_args(parser)
+    return parser
+
+
+def options(parser):
+    parser._optionals.title = "options"
+    meg = parser.add_mutually_exclusive_group(required=True)
+    meg.add_argument(
+        "--seq-path",
+        default=None,
+        help="path to a directory containing FASTQ files to use as input; incompatible with --files",
+        metavar="PATH",
+    )
+    meg.add_argument(
+        "--files",
+        default=None,
+        help="a list of FASTQ files to use as input; incompatible with --seq-path",
+        metavar="FQ",
+        nargs="+",
+    )
+    parser.add_argument(
+        "-o",
+        "--outfile",
+        default="config.cfg",
+        help='output config file; by default, "config.cfg"',
+        metavar="FILE",
+        type=str,
+    )
```

Comment:
Reiterating my suggestion that stdout is used as default.

```python
    parser.add_argument(
        "-o",
        "--outfile",
        default=sys.stdout,
        help='output config file; by default, output is written to terminal (standard output)',
        metavar="FILE",
        type=argparse.FileType("w"),
    )
```

The `.write_config_file()` method would need to be updated to take a file handle instead of a string, and the main method to do the directory handling, but that's probably more appropriate anyway given the scope of these units of code.

Reply:
Completely missed the previous comment. I agree that the stdout is the way to go. Simplified the code significantly and users can pipe the output into a file, etc.